source/mir/math/stat.d

```diff
@@ -424,40 +424,48 @@ unittest
     assert(float.max.repeat(3).hmean!(double, "fast").approxEqual(float.max));
 }
 
-private template centerImpl(F)
-{
-    import mir.ndslice.slice: Slice, SliceKind;
-
-    @safe pure nothrow @nogc
-    auto centerImpl(Iterator, size_t N, SliceKind kind)(
-        Slice!(Iterator, N, kind) slice, F average)
-    {
-        return slice - average;
-    }
-}
-
 /++
-Centers `r`, which must be a finite iterable.
+Centers `slice`, which must be a finite iterable.
 
-By default, `r` is centered by the mean. A custom function may also be provided
+By default, `slice` is centered by the mean. A custom function may also be provided
 using `centralTendency`.
 
 Returns:
-    The elements in the range r with the average subtracted from them.
+    The elements in the slice with the average subtracted from them.
 +/
 
 /// ditto
 template center(alias centralTendency = mean!(Summation.appropriate))
 {
+    import mir.ndslice.slice: Slice, SliceKind, sliced, hasAsSlice;
     /++
     Params:
-        r = range
+        slice = slie
     +/
     @safe pure nothrow
```

Comment:
explicit attributes should be removed because user-provided slice may not provide the attributes.